### PR TITLE
New option in files_external:list --mount-options

### PR DESCRIPTION
**Pull request by mmattel**
## Description

Adding a new option for the `files_external:list` command `--mount-options`.
Using `--mount-options` shows all mount options independent the fact if they are default or not.
I came across when working on #36397 where I wanted to see if an option is set or not.

```
./occ files_external:list  --output=json_pretty
[
    {
        "mount_id": 2,
        "mount_point": "\/ttt",
        "storage": "SMB \/ CIFS",
        "authentication_type": "Username and password",
        "configuration": "host: \"filer\", share: \"swwindows\", root: \"\", domain: \"mydomain\", user: \"martin\", password: \"***\"",
        "options": "",
        "applicable_users": "All",
        "applicable_groups": ""
    }
]
```
```
./occ files_external:list  --mount-options --output=json_pretty
[
    {
        "mount_id": 2,
        "mount_point": "\/ttt",
        "storage": "SMB \/ CIFS",
        "authentication_type": "Username and password",
        "configuration": "host: \"filer\", share: \"swwindows\", root: \"\", domain: \"mydomain\", user: \"martin\", password: \"***\"",
        "options": "encrypt: true, previews: true, filesystem_check_changes: 1, enable_sharing: false, encoding_compatibility: false",
        "applicable_users": "All",
        "applicable_groups": ""
    }
]
```
**NOTE:** during testing, I found a bug in the  `json`/ `json_pretty` output when using option `--all` causing a php error. This is now fixed.

## Motivation and Context

Helps identifying which options are present and in which state independent if they are set to default or not.

## How Has This Been Tested?

Running local

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: will be made on merge 
- [x] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by mmattel** [comment]
CI is :nauseated_face:

File: changelog/unreleased/36420-1 [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Bug: Fix a php error of occ command files_external:list --output=json respectively --output=json_pretty
```
Comment (tboerger) [inline changelog/unreleased/36420-1]: Fixes should not be put together with a new feature.

Comment (mmattel) [reply changelog/unreleased/36420-1]: You are right. It was not intended to find this issue, but is was a very easy win I took with only a minimum of changes.

File: changelog/unreleased/36420 [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Enhancement: Adding a new option in occ command files_external:list --mount-options
```
Comment (micbar) [inline changelog/unreleased/36420]: @mmattel The title should not exceed 80 chars

File: changelog/unreleased/36420-1 (hunk 1 above)
Comment (micbar) [reply changelog/unreleased/36420-1]: Same as above

File: apps/files_external/lib/Command/ListCommand.php [hunk 3]
```diff
@@ -81,6 +81,11 @@ protected function configure() {
 				null,
 				InputOption::VALUE_NONE,
 				'Show passwords and secrets'
+			)->addOption(
+				'mount-options',
+				null,
+				InputOption::VALUE_NONE,
+				'Show all mount options even they are set to default'
```
Comment (micbar) [inline apps/files_external/lib/Command/ListCommand.php]: missing "if"

**Review by micbar — CHANGES_REQUESTED** [review]

**Comment by mmattel** [comment]
Ping ?

**Review by micbar — APPROVED** [review]

**Pull request merged**